docs/docsite/rst/contributor/DJANGO_REQUIREMENTS.rst

```diff
@@ -0,0 +1,843 @@
+                    f"to {request.path} through the API"
+                )
+            return ret
+
+5.3 Custom Permission Classes
+-----------------------------
+
+**REQUIRED**: Implement comprehensive permission checking:
+
+.. code-block:: python
+
+    # api/permissions.py
+    class ModelAccessPermission(permissions.BasePermission):
+        """Model-based access control with hierarchy support"""
+
+        def has_permission(self, request, view):
+            if hasattr(view, 'parent_model'):
+                parent_obj = view.get_parent_object()
+                return check_user_access(
+                    request.user,
+                    view.parent_model,
+                    'read',
+                    parent_obj
+                )
+            return True
+
+**Requirements**:
+
+- Multiple authentication methods (JWT, Session, Basic)
+- Custom pagination, renderers, and metadata classes
+- Comprehensive API exception handling
+- Resource-based URL organization
+- Logging for authentication events
+
+----
+
+6. Security Requirements
+========================
+
+6.1 Production Security Settings
+--------------------------------
+
+**REQUIRED**: Enforce secure defaults for production:
+
+.. code-block:: python
+
+    # settings/production.py
+    DEBUG = False
+    SECRET_KEY = None  # Force explicit configuration
+    ALLOWED_HOSTS = []  # Must be explicitly set
+
+    # Session security
+    SESSION_COOKIE_SECURE = True
+    SESSION_COOKIE_HTTPONLY = True
+    SESSION_COOKIE_SAMESITE = 'Lax'
+    SESSION_COOKIE_AGE = 1800
+
+    # CSRF protection
+    CSRF_COOKIE_SECURE = True
+    CSRF_COOKIE_HTTPONLY = True
+    CSRF_TRUSTED_ORIGINS = []
+
+6.2 External Secret Management
+------------------------------
+
+**REQUIRED**: Implement external secret loading:
+
+.. code-block:: python
+
+    # Secret key from external file
+    SECRET_KEY_FILE = os.environ.get('SECRET_KEY_FILE', '/etc/awx/SECRET_KEY')
+    if os.path.exists(SECRET_KEY_FILE):
+        with open(SECRET_KEY_FILE, 'rb') as f:
+            SECRET_KEY = f.read().strip().decode()
+    else:
+        if not DEBUG:
+            raise ImproperlyConfigured("SECRET_KEY must be configured in production")
+
+6.3 Proxy and Network Security
+------------------------------
+
+**REQUIRED**: Configure reverse proxy security:
+
+.. code-block:: python
+
+    # Proxy configuration
+    REMOTE_HOST_HEADERS = ['REMOTE_ADDR', 'REMOTE_HOST']
+    PROXY_IP_ALLOWED_LIST = []
+    USE_X_FORWARDED_HOST = True
+    USE_X_FORWARDED_PORT = True
+    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+**Requirements**:
+
+- External secret file management
+- Secure cookie configuration
+- CSRF protection with trusted origins
+- Proxy header validation
+- Force HTTPS in production
+
+----
+
+7. Database Management
+======================
+
+7.1 Advanced Database Configuration
+-----------------------------------
+
+**REQUIRED**: Optimize database connections for production:
```

Comment:
```suggestion
**REQUIRED**: Robust database connections for production:
```

The keepalive settings are more about a robust connection.
Atomic_requests .. also robust.
conn_max_age .. robust.
postgres .. it's what we support .. it's robust.